libs/content/workspace-ffi/SwiftWorkspace/Sources/Editor/iOSMTK.swift

```diff
@@ -163,39 +163,34 @@ public class iOSMTKTextInputWrapper: UIView, UITextInput, UIDropInteractionDeleg
             self.mtkView.setNeedsDisplay()
         }
     }
+    
+    func sendImage(img: Data, isPaste: Bool) {
+        let imgPtr = img.withUnsafeBytes { (pointer: UnsafeRawBufferPointer) -> UnsafePointer<UInt8> in
+            return pointer.baseAddress!.assumingMemoryBound(to: UInt8.self)
+        }
+        
+        clipboard_send_image(wsHandle, imgPtr, UInt(img.count), isPaste)
+    }
 
-    func importContent(_ importFormat: SupportedImportFormat) -> Bool {
+    func importContent(_ importFormat: SupportedImportFormat, isPaste: Bool) {
         switch importFormat {
         case .url(let url):
-            if let markdownURL = workspaceState!.importFile(url) {
-                pasteText(text: markdownURL)
+            if url.pathExtension.lowercased() == "png" {
+                guard let data = try? Data(contentsOf: url) else {
+                    return
+                }
                 
-                return true
+                sendImage(img: data, isPaste: isPaste)
+            } else {
+                clipboard_send_file(wsHandle, url.path(percentEncoded: false), isPaste)
             }
         case .image(let image):
-            if let data = image.pngData() ?? image.jpegData(compressionQuality: 1.0),
-               let url = createTempDir() {
-                let imageUrl = url.appendingPathComponent(String(UUID().uuidString.prefix(10).lowercased()), conformingTo: .png)
-
-                do {
-                    try data.write(to: imageUrl)
-                } catch {
-                    return false
-                }
-
-                if let lbImageURL = workspaceState!.importFile(imageUrl) {
-                    pasteText(text: lbImageURL)
-
-                    return true
-                }
+            if let img = image.pngData() ?? image.jpegData(compressionQuality: 1.0) {
```

Comment:
did you test with each kind of image? would be somewhat surprised if jpegs worked

Reply:
This was some old code from before, and I believe I tested it with tiff, jpeg, and png (back then at least). It works right now too, I just tried jpegs.